contributors/guide/owners.md

```diff
@@ -49,6 +51,22 @@ reviewers:
   - sig-foo # this is an alias
 ```
 
+#### Filters
+
+An OWNERS file may also include a `filters` key.
+The `filters` key is a map whose keys are regular expressions and whose values are [simple OWNERS configurations](#owners).
```

Comment:
It would be nice if we explained the paths that the regular expressions are matched against.  Are they absolute?  Relative to the repo root?  Relative to the `OWNERS` file they came from?  I'm having a hard time figuring this out by reading the source.   @cjwagner, @BenTheElder, do either of you remember off the top of your head?

Reply:
Ah, I didn't know the regexp filtering was documented anywhere. We had intentionally left it undocumented, because it is not handled properly by the `approve` plugin (and possibly `blunderbuss`).  The `approve` plugin will need a significant rework to work well with regexp filtering.

Reply:
https://github.com/kubernetes/test-infra/issues/7690

Reply:
> Ah, I didn't know the regexp filtering was documented anywhere.

It's not.  This PR was my attempt at documenting it, after I ran across `filters` in the wild and wondered what it was ;).

> We had intentionally left it undocumented...

Hmm.  Maybe we should hold on this PR until that gets fixed?  Or land this documenting the limitation?

Do you know the answer to my path question?

Reply:
> Maybe we should hold on this PR until that gets fixed? Or land this documenting the limitation?

I think the later makes the most sense. For now the filters should only be used with the labels section. 

> Do you know the answer to my path question?

The paths are relative to the OWNERS file they are in.


We should also note that the `filters` section is mutually exclusive with the normal simple OWNERS config at the root of the file.

Reply:
> We should also note that the filters section is mutually exclusive with the normal simple OWNERS config at the root of the file.

How does that combine with the `approve` limitation?  Does it mean that an `OWNERS` file which sets `filters` should currently only set `labels`, and that you shouldn't do something like:

```yaml
approvers:
- alice
filters:
  "\\.go$":
    labels:
    - re/go
```

Reply:
They are unrelated limitations.
 The first is a formatting limitation. You can't specify a simple config if you specify `filters`. You should instead specify an entry in `filters` for the `.*` key. So your example would still be invalid even if it was:
```
labels:
- foo
filters:
  "\\.go$":
    labels:
    - re/go
```
The second is a limitation due to bugs. You should only set the `labels` section if you are using `filters`. So this is invalid:
```
filters:
  "\\.go$":
    approvers:
    - alice
```